packages/block-library/src/archives/block.json

```diff
@@ -0,0 +1,32 @@
+{
+	"name": "core/archives",
+	"title": "Archives",
+	"category": "widgets",
+	"icon": {
+		"src": "icon.js"
```

Comment:
This one kind of sticks out as far as being inconsistent with other file references like in `edit` and `renderCallback`. I see from #13693 this is done as such to support properties like `foreground` and `background` color assignments.

In all, I think it's okay, though I also wonder on a few variations:

- Optionally also allow `icon` to be assigned as a string if it only needs to assign `src`
- Create a pattern around "file reference with options", something like a tuple of `[ file, options ]` (`{ "icon": [ "icon.js", { "background": "red" } ] }`)

Was looking at some similar specifications, where `src` is a pretty accepted convention for naming this type of resource ([example](https://w3c.github.io/manifest/#src-member)).

Reply:
> It is reserved for the slug at the moment, which aligns with how you can register block's icon as of today. I would be happy to use the file name as a shortcut but then we need to figure out how to provide the slug differently.

Oh right. That makes sense. I guess the other option is to inspect the shape of the string to see if it "appears" as a file path (e.g. has file extension, or starts with `./`)